src/hooks/useCirclePost.ts

```diff
@@ -0,0 +1,139 @@
+import { useCallback } from 'react';
+import { useQuery } from 'react-query';
+import { useActiveAccount } from './useActiveAccount';
+import { gql } from 'graphql-request';
+import { useGraphQLClient } from './useGraphQLClient';
+
+export const useCirclePost = (postId: string, disabled?: boolean) => {
+  const { graphQLClient } = useGraphQLClient();
+  const { isFetched, accountHeaders } = useActiveAccount();
+
+  const queryForPostDetails = useCallback(async () => {
+    return graphQLClient.request<PostDetailsPostQueryResponse, { id: string }>(
+      postDetailsQuery,
+      {
+        id: postId,
+      },
+      accountHeaders,
+    );
+  }, [accountHeaders, graphQLClient, postId]);
+
+  return useQuery('postDetails', queryForPostDetails, {
+    enabled: !disabled && isFetched && !!accountHeaders,
+  });
+};
+
+export type Post = {
+  __typename: string;
+  author: {
+    profile: {
+      displayName: string;
+    };
+  };
+  id: string;
+  parentId: string;
+  priority: string;
+  replyCount: number;
+  createdAt: string;
+  deletedAt?: string;
+  status: string;
+  message: string;
+  attachments: {}[];
+  reactionTotals: {
+    type: string;
+    url: string;
+    count: number;
+    userHasReacted: boolean;
+  }[];
+  metadata: unknown;
+  replies: {
+    edges: {
+      node: Post;
+    }[];
+    pageInfo: {
+      endCursor?: string;
+      hasNextPage?: boolean;
+    };
+  };
+  circle?: {
```

Comment:
I _think_ it would be cleaner to only query for fields that have an immediate use and expand as we go to avoid unnecessary bloat but perhaps you already have a use for these fields mapped out.

Reply:
Yeah good call. That was an artifact from an earlier iteration. I'll remove it.